ui/src/components/loading.tsx

```diff
@@ -38,12 +41,24 @@ export default function Loading(props: Props) {
     return <>{props.children}</>;
   }
 
-  return <Box className="loading">{showStatus(visible, props.status)}</Box>;
+  return (
+    <Box className={props.size === "small" ? "" : "loading"}>
+      {showStatus(visible, props.status, props.size)}
+    </Box>
+  );
 }
 
-function showStatus(visible: boolean, status?: Status): ReactNode {
+function showStatus(
+  visible: boolean,
+  status?: Status,
+  size?: string
+): ReactNode {
   if (status?.error && !status?.isPending) {
-    return (
+    return size === "small" ? (
```

Comment:
Ternary operators that span more than a couple lines are difficult to read so avoid them when it's reasonable to do so. In this case it's easy:
```
if (size === "small") {
  return ...;
}
return ...;
```